File: Tercer trimestre/Clases/Clase3.3/Polimorfismo.py
# clase base 
class Animal:
    def hablar(self):
        print("el animal hace un ruido ")

#clase hija 1 
class Perro(Animal):
    def hablar(self):
        print("el perro dice; ¡Guua guua!")

#clase hija 2 
class Gato(Animal):
    def hablar(self):
        print("el Gato dice; ¡Miauu miauu!")


#funcion que demuestra el polimorfismo 

def hacer_hablar(animal):
    animal.hablar()

#crear el objeto 

perro = Perro()
gato = Gato()

# Usar esta misma funcion con diferentes tipos de objetos 
hacer_hablar(perro)
hacer_hablar(gato)

File: Tercer trimestre/Clases/Clase3.3/polimorfismo.java
// clase base
class Animal {
    public void hablar(){
        System.out.println("el animal hace un ruido ");
    }    
}

// classe hija 1 
class Perro extends Animal{
    @Override
    public void hablar(){
         System.out.println("el perro dice; ¡Guau guau!");
        }
}

// classe hija 2 
class Gato extends Animal{
    @Override //Indica la herencia del metodo hablar de la clase base Animal
    public void hablar(){
         System.out.println("el Gato dice; ¡Miau miau!");
        }
}

// clase principal para ejecutar este programa
public class Main{
    //metodo que demuestre el polimorfismo
    public static void hacerHablar(Animal animal){
        animal.hablar();
    }

    public static void main(String[] args) {
        Perro perro = new Perro();
        Gato gato = new Gato();

        // Usar esta misma funcion con diferentes tipos de objetos
        hacerHablar(perro);
        hacerHablar(gato);
    }
}